### DISCUSSION TITLE
Setting Retention Policy

### DISCUSSION THREAD

**Discussion by klaushbrk1** [datalust/seq-tickets#1380] [Help]
Hi there,
when I try to add/set a retention policy from the seq web front end, I get the following error:
"Could not save retention policy details. An unhandled error occurred while serving the request." 

When I look into the seq diagnostics log, I can see the following problem (MDB_PANIC):
Any help would be much appreciated.

`<snip>`

**Comment by nblumhardt** [comment]
Thanks for getting in touch! I edited your post to remove some info about your Seq server's hostname.

This looks like the kind of thing we see when Seq is run on an Azure Files share; any chance that's the environment your server is running in?

If not, some environment info would be useful; it would also be great to know whether restarting the Seq container resolves this.

Thanks!

**Reply by klaushbrk1** [reply]
Thanks for getting that quick in touch! Much appreciated.
We are running seq in in docker container on a Linux Ubuntu host (internal company network) and it has a file share to a windows network drive. We can successfully log messages, see them in seq, i.e. it should not be a problem accessing that shared windows drive (?).

**Reply by klaushbrk1** [reply]
Where does seq actually store it's internal configuration (like the retention policies)?

**Reply by nblumhardt** [reply]
Thanks for the follow-up! The network share isn't going to play nicely, unfortunately.

Seq stores configuration in `/data/Documents`, in an LMDB database. Most similar storage engines (including the one Seq uses for log storage under `/data/Stream`) need very precise guarantees around flushing of dirty files/pages to disk, and it seems that network filesystem drivers tend to fudge these things for performance reasons, leading to data corruption.

A local disk attached to the host is your best bet; in conjunction with this, you might also consider moving the configuration over to PostgreSQL (see https://docs.datalust.co/docs/using-postgresql-as-a-metastore for more info).

HTH!

**Reply by klaushbrk1** [reply]
Thanks for those insights. We changed from using direct access to the network share to use a docker volume (but also restarted seq). We can now successfully set the retention policy. We will consider changing over to use PostgreSQL as a configuration store

**Reply by nblumhardt** [reply]
Great 👍
